### PR TITLE
AO3-4702 Test autocomplete

### PR DESCRIPTION
**Pull request by zz9pzza**
# Pull Request Checklist
- [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
- [x] Have you read through the [contributor guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
- [x] Have you added tests for any changed functionality?
- [x] Have you added the JIRA issue number as the _first_ thing in your pull request title (eg: `AO3-1234 Fix thing`)
- [x] Have you updated the JIRA issue with the information below?
## Issue

https://otwarchive.atlassian.net/browse/AO3-4702

## Purpose

Test the autocomplete.

## Testing

No manual testing required.

### CODE REVIEW

File: features/step_definitions/web_steps.rb [hunk 1]
```diff
@@ -33,6 +33,13 @@ def with_scope(locator)
   page.driver.browser.url_whitelist = ['http://127.0.0.1']
 end
 
+When /^I type in "([^"]*)" with "([^"]*)"$/ do |selector,contents|
+ find(selector).native.send_keys(contents)
+ #sleep 5
+ #puts page.evaluate_script( 'document.documentElement.innerHTML' )
+ puts find(:css , 'div.autocomplete.dropdown', :wait=> 20 , :text => 'uby' ).text
```
Comment (sarken) [inline features/step_definitions/web_steps.rb]: For what it's worth, the text that autocomplete generates is actually inside a li inside a ul inside the div:

```
<div class="autocomplete dropdown" style="position: absolute; z-index: 999; display: none;">
  <ul role="listbox" aria-activedescendant="ui-active-menuitem" style="display: block;">
    <li role="option" class="odd"><b>Te</b>sting</li>
```

So I don't think looking for it in the div will work -- I think you'll have to look at the list items themselves with something like `div.autocomplete li`

File: features/tags_and_wrangling/tag_wrangling_fandoms.feature [hunk 2]
```diff
@@ -211,3 +211,25 @@ Scenario: fandoms wrangling - syns, mergers, autocompletes, metatags
...
+        | name                            | type         | canonical |
+        | Steven Universe                 | Fandom       | true      |
+        | Ruby (Pokemon)                  | Character    | true      |
+        | Ruby (Firefly)                  | Character    | true      |
+        | Ruby (Steven Universe)          | Character    | true      |
+        | Ruby (SU)                       | Character    | false     |
+    And I am logged in as "Enigel" with password "wrangulate!"
+    When I go to the "Ruby (SU)" tag page
+     And I follow "Edit"
+     And I type in "ul.autocomplete" with "Ruby"
```
Comment (bingeling) [inline features/tags_and_wrangling/tag_wrangling_fandoms.feature]: "ul.autocomplete" is the container, "ul.autocomplete input" is the form field to type into

**Comment by sarken** [comment]
Is this still a WIP?

**Comment by sarken** [comment]
This has conflicts and will need to be updated, but it's still marked "WIP," so I'm not sure if it's done.

**Comment by shalott** [comment]
I hate to say this, but I think this is fundamentally a bad idea as well as a hack. You are not actually testing the autocomplete, you're just creating the illusion of having tested it, which IMO is worse because it gives us false confidence. 

If you've got the testing-with-javascript working, I suspect the issue is you were not looking at the right field -- the autocomplete javascript does a thing where it replaces one div with another which holds the actual results. I actually wrote code that I think was working to access it but never managed to finish implementing the PR involved, I will dig the code out if you want?

**Comment by zz9pzza** [comment]
I had high hopes for:

assert page.should have_css('div.autocomplete.dropdown', :text => answer)

but no

**Comment by sarken** [comment]
Closing in favor of #2877

**Pull request closed without merging**